app/controllers/admin/budgets_controller.rb

```diff
@@ -14,10 +14,15 @@ def show
     @budget = Budget.includes(groups: :headings).find(params[:id])
   end
 
-  def new
-  end
+  def new; end
+
+  def edit; end
+
+  def calculate_winners
+    @budget.headings.each { |heading| Budget::Result.new(@budget, heading).delay.calculate_winners }
```

Comment:
We're using DelayedJob here, since the `calculate_winners` hasn't been tested for its performance on production.... And its easy to imagine that a ton of Headings with a ton of Investments could draw some power from the server if not run in the background :)